### DISCUSSION TITLE
Trace ID usage

### DISCUSSION THREAD

**Discussion by ghost** [open-telemetry/opentelemetry-specification#2735] [General]
My question is vague. I read that the trace ID is 128 bit (https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md). Does this mean that this is enough permutations to be sure, it's not gonna generate the same trace ID ever?

I'm trying to figure out, can I use the Otel's generated trace ID as something that you can bind to an "operation" in a business context. My concern is that the Otel's trace ID is not suitable for that. I also know about UUID (https://www.rfc-editor.org/rfc/rfc4122) which seems more serious for generating IDs.

Would you not recommend to use Trace ID for something outside of "telemetry"?

Can you comment anything on that? Thank you.

**Comment by arminru** [comment]
Hi @smthlow!
What you are describing is not what the Trace ID is intended for. It is based on the W3C TraceContext specification, which you can read up more on https://w3c.github.io/trace-context/.
Note that while both a UUID and a Trace ID contain of 128 bits, the TraceContext specification only imposes a SHOULD requirement for its randomness, and for only 56 of the bits a MUST requirement for randomness exists, if the `random` flag is set.
See https://w3c.github.io/trace-context/#trace-id and https://w3c.github.io/trace-context/#randomness-of-trace-id in particular.
Since this lower entropy increases the risk of collisions, it would be safer to rely on UUIDs, but you might still find the 56 bits of entropy acceptable for your application.

**Reply by Oberon00** [reply]
This flag seems to be something rather new, I don't think OTel supports that draft version.

**Reply by ghost** [reply]
Thanks for the reply.
